processor/src/trace/mod.rs

```diff
@@ -293,6 +308,18 @@ where
         "trace length must be at least {MIN_TRACE_LEN}, but was {trace_len}",
     );
 
+    // get the lengths of the traces: main, range, and chiplets
+    let trace_len_summary = TraceLenSummary::new(
+        clk as usize,
+        range_table_len,
+        ChipletsLengths::new(
+            chiplets.bitwise_start(),
+            chiplets.memory_start() - chiplets.bitwise_start(),
+            chiplets.kernel_rom_start() - chiplets.memory_start(),
+            chiplets.padding_start() - chiplets.kernel_rom_start(),
+        ),
```

Comment:
Should we modify `ChipletsLengths::new()` method so that it just takes `chiplets` as a parameter and does all this computations internally?